dhis-2/dhis-services/dhis-service-acl/src/main/java/org/hisp/dhis/security/acl/DefaultAclService.java

```diff
@@ -578,12 +579,21 @@ private boolean haveOverrideAuthority( User user )
 
     private boolean canAccess( User user, Collection<String> anyAuthorities )
     {
-        return haveOverrideAuthority( user ) || anyAuthorities.isEmpty() || haveAuthority( user, anyAuthorities );
+        if ( user == null )
+        {
+            return true;
+        }
+        boolean a = haveOverrideAuthority( user );
+        boolean b = anyAuthorities.isEmpty();
+        boolean c = haveAuthority( user, anyAuthorities );
+        return a || b || c;
     }
 
     private boolean haveAuthority( User user, Collection<String> anyAuthorities )
     {
-        return containsAny( user.getUserCredentials().getAllAuthorities(), anyAuthorities );
+        Set<String> allAuthorities = user.getUserCredentials().getAllAuthorities();
+        boolean a = containsAny( allAuthorities, anyAuthorities );
```

Comment:
Not sure why we need to rewrite this?

Reply:
Wrongly committed code used for debugging.